### PR TITLE
Google docstring conversion batch 4

### PR DESCRIPTION
**Pull request by michael-berk**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/michael-berk/mlflow/pull/10831?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10831/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10831
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?
Google docstring conversion for tracking directory.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/tracking/_tracking_service/utils.py [hunk 1]
```diff
@@ -37,20 +37,18 @@ def is_tracking_uri_set():
...
 
 def set_tracking_uri(uri: Union[str, Path]) -> None:
-    """
-    Set the tracking server URI. This does not affect the
+    """Set the tracking server URI. This does not affect the
     currently active run (if one exists), but takes effect for successive runs.
 
-    :param uri:
-
-                - An empty string, or a local file path, prefixed with ``file:/``. Data is stored
```
Comment (michael-berk) [inline mlflow/tracking/_tracking_service/utils.py:46]: TODO

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -138,19 +141,18 @@ def _get_registry_client(self):
 
     def get_run(self, run_id: str) -> Run:
         """
-        Fetch the run from backend store. The resulting :py:class:`Run <mlflow.entities.Run>`
```
Comment (michael-berk) [inline mlflow/tracking/client.py:141]: TODO

File: mlflow/tracking/client.py [hunk 3]
```diff
@@ -2117,122 +1906,52 @@ def print_registered_model_info(rm):
...
+        Args:
+            name: Name of the registered model to update.
+            new_name: New proposed name for the registered model.
+
+        Returns:
+            A single updated :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        self._get_registry_client().rename_registered_model(name, new_name)
+
+    def update_registered_model(
```
Comment (michael-berk) [inline mlflow/tracking/client.py:1925]: TODO: check `def update_registered_model` in `mlflow/tracking/client.py`

**Comment by michael-berk** [comment]
Rendered local html build diff
```
/Users/michael.berk/dev/mlflow/google-docs/batch_4/mlflow/docs/build/html/python_api/mlflow.html
------------------------------
1822: -                  – The experiment name, which must be unique and is case sensitive
1823: +                  – The experiment name, which must be unique and is case sensitive.
1832: -                  – The location to store run artifacts.
1833: +                  – The location to store run artifacts. If not provided, the server picks
1835: - If not provided, the server picks an appropriate default.
1836: + an appropriate default.
1844: -                  – An optional dictionary of string keys and values to set as
1845: +                  – An optional dictionary of string keys and values to set as tags on the experiment.
1847: - tags on the experiment.
1858: -                String ID of the created experiment.
1859: +                 String ID of the created experiment.
1869: -                Example
1870: +                   Example
1987: -                – The The string-ified experiment ID returned from
1989: +                – The string-ified experiment ID returned from
2128: - print(
2129: -     f"run_id: {run_id}; lifecycle_stage: {mlflow.get_run(run_id).info.lifecycle_stage}"
2131: + lifecycle_stage = mlflow.get_run(run_id).info.lifecycle_stage
2133: - )
2134: + print(f"run_id: {run_id}; lifecycle_stage: {lifecycle_stage}")
4872: - If
4875: +              If
5452: -                Example
5476: -                Output
5544: +              Fetch the run from backend store. The resulting Run contains a collection of run metadata –
5545: -              Fetch the run from backend store. The resulting
5549: -                 Run
5553: -              contains a collection of run metadata –
5557: -                 RunInfo
5561: -              ,
5562: - as well as a collection of run parameters, tags, and metrics –
5563: + RunInfo as well as a collection of run parameters, tags, and metrics – RunData. It also
5568: -                 RunData
5572: -              . It also contains a collection of run
5573: - inputs (experimental), including information about datasets used by the run –
5575: + contains a collection of run inputs (experimental), including information about datasets used by
5577: + the run – RunInputs. In the case where multiple metrics with the same key are logged for the
5581: -                 RunInputs
5585: -              . In the case where multiple metrics with the
5586: - same key are logged for the run, the
5590: -                 RunData
5594: -              contains the
5595: - most recently logged value at the largest step for each metric.
5596: + run, the RunData contains the most recently logged value at the largest step for each metric.
5616: -                A single
5620: -                   mlflow.entities.Run
5624: -                object, if the run exists. Otherwise,
5625: +                A single Run object, if the run exists. Otherwise, raises an exception.
5627: - raises an exception.
5727: -                Example
5746: -                Output
7439: -               Returns
7443: -                None
7680: -                  – Metric name (string). This string may only contain alphanumerics, underscores (_),
7682: +                  – Metric name. This string may only contain alphanumerics, underscores (_),
7694: -                  – Metric value (float). Note that some special values such as +/- Infinity may be
7696: +                  – Metric value. Note that some special values such as +/- Infinity may be
7708: -                  – Metric step (int). Defaults to zero if unspecified.
7710: +                  – Metric step. Defaults to zero if unspecified.
8150: -                  – Parameter name (string). This string may only contain alphanumerics,
8152: +                  – Parameter name. This string may only contain alphanumerics, underscores (_), dashes
8154: + (-), periods (.), spaces ( ), and slashes (/). All backend stores support keys up to
8155: + length 250, but some may support larger keys.
8156: - underscores (_), dashes (-), periods (.), spaces ( ), and slashes (/).
8157: - All backend stores support keys up to length 250, but some may
8158: - support larger keys.
8166: -                  – Parameter value (string, but will be string-ified if not).
8168: +                  – Parameter value, but will be string-ified if not. All built-in backend stores support
8170: + values up to length 6000, but some may support larger values.
8171: - All built-in backend stores support values up to length 6000, but some
8172: - may support larger values.
8184: -                  If True, blocks until the parameter is logged
8185: +                  If True, blocks until the parameter is logged successfully. If
8187: - successfully. If False, logs the parameter asynchronously and
8188: - returns a future representing the logging operation.
8189: + False, logs the parameter asynchronously and returns a future representing the logging
8190: + operation.
8216: -                instance that
8217: +                instance that represents
8219: - represents future for logging operation.
8221: + future for logging operation.
8597: -               Returns
8601: -                None
9080: - Note that this method assumes the model registry backend URI is the same as that of the
9081: +              Note that this method assumes the model registry backend URI is the same as that of the
10197: -                   ), defaults to searching for all
10199: +                   ), defaults to
10200: - experiments. The following identifiers, comparators, and logical operators are supported.
10202: + searching for all experiments. The following identifiers, comparators, and logical
10204: + operators are supported.
10444: -                   , so
10446: +                   ,
10447: + so
10453: -                   is
10454: +                   is equivalent to
10456: - equivalent to
10474: -                   ,
10475: - which lists experiments updated most recently first. The following fields are supported:
10476: +                   , which lists experiments updated most recently first.
10477: + The following fields are supported:
11012: -                A list of
11013: +                  A list of
11020: -                   mlflow.entities.model_registry.ModelVersion
11021: +                     mlflow.entities.model_registry.ModelVersion
11026: +                  objects
11032: -                objects
11033: - that satisfy the search expressions.
11034: +                   that satisfy the search expressions.
11302: +                   Filter query string (e.g., “name = ‘a_model_name’ and tag.key = ‘value1’”),
11303: -                   Filter query string
11304: - (e.g.,
11307: -                     "name
11310: -                     =
11313: -                     'a_model_name'
11316: -                     and
11319: -                     tag.key
11322: -                     =
11325: -                     'value1'"
11328: -                   ),
11342: -                         name
11345: -                       : registered model name.
11346: +                       ”name”: registered model name.
11354: -                         tags.<tag_key>
11357: -                       : registered model tag. If
11362: -                         tag_key
11365: -                       contains spaces, it must be
11366: - wrapped with backticks (e.g.,
11369: +                       ”tags.<tag_key>”: registered model tag. If “tag_key” contains spaces, it must be
11370: + wrapped with backticks (e.g., “tags.`extra key`”).
11371: -                         "tags.`extra
11374: -                         key`"
11377: -                       ).
11391: -                         =
11394: -                       : Equal to.
11395: +                       ”=”: Equal to.
11403: -                         !=
11410: -                       : Not equal to.
11411: +                       ”!=”: Not equal to.
11419: -                         LIKE
11426: -                       : Case-sensitive pattern match.
11427: +                       ”LIKE”: Case-sensitive pattern match.
11435: -                         ILIKE
11442: -                       : Case-insensitive pattern match.
11443: +                       ”ILIKE”: Case-insensitive pattern match.
11458: -                         AND
11461: -                       : Combines two sub-queries and returns True if both of them are True.
11462: +                       ”AND”: Combines two sub-queries and returns True if both of them are True.
12262: -                If output_format is
12265: -                  list
12268: -                : a list of
12270: +                a list of
12316: +               Return type
12320: -                Example
12327: +                 If output_format is
12330: +                   list
12337: +                   Example
12630: - Lifecycle_stage: active
12632: + Lifecycle_stage: activ
12721: -                  – Tag name (string). This string may only contain alphanumerics, underscores
12723: +                  – Tag name. This string may only contain alphanumerics, underscores (_), dashes (-),
12725: + periods (.), spaces ( ), and slashes (/). All backend stores will support keys up to
12726: + length 250, but some may support larger keys.
12727: - (_), dashes (-), periods (.), spaces ( ), and slashes (/).
12728: - All backend stores will support keys up to length 250, but some may
12729: - support larger keys.
12737: +                  – Tag value, but will be string-ified if not. All backend stores will support values
12738: + up to length 5000, but some may support larger values.
12739: -                  – Tag value (string, but will be string-ified if not).
12740: - All backend stores will support values up to length 5000, but some
12741: - may support larger values.
12944: -                –
12948: -                   An empty string, or a local file path, prefixed with
12950: +                – An empty string, or a local file path, prefixed with
12954: +                  file:/
12959: -                     file:/
12962: -                   . Data is stored
12964: +                . Data is stored
12970: -                     ./mlruns
12972: +                  ./mlruns
12975: -                   if empty).
12982: -                   An HTTP URI like
12984: +                if empty). An HTTP URI like
12990: -                     https://my-tracking-server:5000
12992: +                  https://my-tracking-server:5000
12995: -                   .
13002: -                   A Databricks workspace, provided as the string “databricks” or, to use a
13004: +                . A Databricks workspace, provided as the string
13006: - Databricks CLI
13007: + “databricks” or, to use a Databricks CLI
13011: -                    profile
13013: +                 profile
13017: -                   ,
13019: +                ,
13314: -              Set a tag under the current run. If no run is active, this method will create a
13315: +              Set a tag under the current run. If no run is active, this method will create a new active
13317: - new active run.
13318: + run.
13331: -                  – Tag name (string). This string may only contain alphanumerics, underscores
13333: +                  – Tag name. This string may only contain alphanumerics, underscores (_), dashes (-),
13335: + periods (.), spaces ( ), and slashes (/). All backend stores will support keys up to
13336: + length 250, but some may support larger keys.
13337: - (_), dashes (-), periods (.), spaces ( ), and slashes (/).
13338: - All backend stores will support keys up to length 250, but some may
13339: - support larger keys.
13347: +                  – Tag value, but will be string-ified if not. All backend stores will support values
13348: + up to length 5000, but some may support larger values.
13349: -                  – Tag value (string, but will be string-ified if not).
13350: - All backend stores will support values up to length 5000, but some
13351: - may support larger values.
13363: -                  If True, blocks until the tag is logged
13364: +                  If True, blocks until the tag is logged successfully. If False,
13366: - successfully. If False, logs the tag asynchronously and
13367: - returns a future representing the logging operation.
13368: + logs the tag asynchronously and returns a future representing the logging operation.
13568: - logs tags asynchronously and returns a future representing the logging
13569: + logs tags asynchronously and returns a future representing the logging operation.
13571: - operation.
13727: +                   . Data is stored locally at the provided file (or
13728: -                   . Data is stored
13729: - locally at the provided file (or
13751: -                   A Databricks workspace, provided as the string “databricks” or, to use a
13752: +                   A Databricks workspace, provided as the string “databricks” or, to use a Databricks CLI
13754: - Databricks CLI
13758: -                   ,
13759: - “databricks://<profileName>”.
13760: +                   , “databricks://<profileName>”.
/Users/michael.berk/dev/mlflow/google-docs/batch_4/mlflow/docs/build/html/python_api/mlflow.client.html
------------------------------
689: -                    – the model URI of the model version to copy. This must be a model
691: +                    – The model URI of the model version to copy. This must be a model
709: -                    – the name of the registered model to copy the model version to. If a
711: +                    – The name of the registered model to copy the model version to. If a
981: -                    – The location to store run artifacts.
982: +                    – The location to store run artifacts. If not provided, the server
984: - If not provided, the server picks an appropriate default.
985: + picks anappropriate default.
1397: -                    – Run ID from MLflow tracking server that generated the model
1398: +                    – Run ID from MLflow tracking server that generated the model.
2191: - This deletion is a soft-delete, not a permanent deletion.
2192: - Experiment names can not be reused, unless the deleted experiment
2193: - is permanently deleted by a database admin.
2196: +                This deletion is a soft-delete, not a permanent deletion. Experiment names can not be
2197: + reused, unless the deleted experiment is permanently deleted by a database admin.
2625: - When stage is set, tag will be deleted for latest model version of the stage.
2626: +                When stage is set, tag will be deleted for latest model version of the stage.
2671: -                 Returns
2675: -                  None
2977: -                 Returns
2981: -                  None
3180: -                 Returns
3184: -                  None
3434: -                    – String ID of the run
3435: +                    – String ID of the run.
3444: -                    – Name of the tag
3445: +                    – Name of the tag.
4335: -                    – Unique identifier for run
4336: +                    – Unique identifier for run.
4345: -                    – Metric name within the run
4346: +                    – Metric name within the run.
4365: -                  entities if logged, else empty list
4366: +                  entities if logged, else empty list.
4517: +                Converts the docstring args and returns to google style.
4732: -                  A single
4733: +                   A single
4740: -                     mlflow.entities.model_registry.ModelVersion
4741: +                      mlflow.entities.model_registry.ModelVersion
4748: -                  object.
4749: +                   object.
4759: -                  Example
4760: +                     Example
4965: -                  Example
5006: -                  Output
5113: +                This is a docstring. Here is info.
5272: -                  object, if the parent run exists.
5273: +                  object, if the parent run exists. Otherwise,
5275: - Otherwise, returns None.
5276: + returns None.
5373: +                Get a registered model.
6049: -                  pandas.DataFrame containing the loaded table if the artifact exists
6050: +                    pandas.DataFrame containing the loaded table if the artifact exists
6052: - or else throw a MlflowException.
6058: +                     or else throw a MlflowException.
6065: -                  Example with passing run_ids
6066: +                     Example with passing run_ids
6771: +                 Raises
6773: -                Raises an MlflowException if any errors occur.
6781: +                    mlflow.MlflowException
6784: +                  – If any errors occur.
7759: -                    – String ID of the run
7760: +                    – String ID of the run.
7783: +                 Raises
7785: -                Raises an MlflowException if any errors occur.
7786: - :return: None
7794: +                    mlflow.MlflowException
7797: +                  – If any errors occur.
8031: -                    – Metric name (string). This string may only contain alphanumerics, underscores
8033: +                    – Metric name. This string may only contain alphanumerics, underscores
8045: -                    – Metric value (float). Note that some special values such
8047: +                    – Metric value. Note that some special values such
8322: -                    – Parameter name (string). This string may only contain alphanumerics, underscores
8324: +                    – Parameter name. This string may only contain alphanumerics, underscores
8336: -                    – Parameter value (string, but will be string-ified if not).
8338: +                    – Parameter value, but will be string-ified if not.
8617: -                 Returns
8621: -                  None
9677: -                     ), defaults to searching for all
9679: +                     ), defaults to
9680: - experiments. The following identifiers, comparators, and logical operators are
9682: + searching for all experiments. The following identifiers, comparators, and logical
9684: - supported.
9685: + operators are supported.
9925: -                     , so
9927: +                     ,
9928: + so
9934: -                     is
9935: +                     is equivalent to
9937: - equivalent to
9955: -                     ,
9956: - which lists experiments updated most recently first. The following fields are supported:
9957: +                     , which lists experiments updated most recently first.
9958: + The following fields are supported:
10362: - defaults to searching for all model versions. The following identifiers, comparators,
10364: + defaults to searching for all model versions. The following identifiers,
10365: - and logical operators are supported.
10366: + comparators, and logical operators are supported.
10418: -                         contains spaces, it must be
10420: +                         contains spaces, it must
10421: - wrapped with backticks (e.g.,
10422: + be wrapped with backticks (e.g.,
10863: +                     Filter query string (e.g., “name = ‘a_model_name’ and tag.key =
10864: -                     Filter query string
10865: - (e.g.,
10868: -                       "name
10871: -                       =
10874: -                       'a_model_name'
10877: -                       and
10880: -                       tag.key
10883: -                       =
10886: -                       'value1'"
10889: -                     ),
10890: - defaults to searching for all registered models. The following identifiers, comparators,
10892: + ‘value1’”), defaults to searching for all registered models. The following
10894: - and logical operators are supported.
10895: + identifiers, comparators, and logical operators are supported.
10926: -                         contains spaces, it must be
10928: +                         contains spaces, it
10929: + must be wrapped with backticks (e.g., “tags.`extra key`”).
10930: - wrapped with backticks (e.g.,
10933: -                           "tags.`extra
10936: -                           key`"
10939: -                         ).
11750: -                  Example
11775: -                  Output
12053: -                 Returns
12057: -                  None
12277: -                 Returns
12281: -                  None
12464: -                 Returns
12468: -                  None
12676: +                    – Tag name. This string may only contain alphanumerics, underscores (_), dashes (-),
12677: + periods (.), spaces ( ), and slashes (/). All backend stores will support keys up to
12678: + length 250, but some may support larger keys.
12679: -                    – Tag name (string). This string may only contain alphanumerics,
12680: - underscores (_), dashes (-), periods (.), spaces ( ), and slashes (/).
12681: - All backend stores will support keys up to length 250, but some may
12682: - support larger keys.
12690: -                    – Tag value (string, but will be string-ified if not).
12692: +                    – Tag value, but will be string-ified if not. All backend stores will support
12694: + values up to length 5000, but some may support larger values.
12695: - All backend stores will support values up to length 5000, but some
12696: - may support larger values.
12708: -                    If True, blocks until the tag is logged successfully.
12709: +                    If True, blocks until the tag is logged successfully. If
12711: - If False, logs the tag asynchronously and returns a future
12713: + False, logs the tag asynchronously and returns a future representing the logging
12715: - representing the logging operation.
12716: + operation.
12738: -                     mlflow.utils.async_logging.run_operations.RunOperations
12740: +                   mlflow.utils.async_logging.run_operations.RunOperations
12746: -                  instance that
12747: +                  instance that represents
12749: - represents future for logging operation.
12751: + future for logging operation.
12958: +                    . Defaults to “FINISHED”.
12959: -                    .
12960: - Defaults to “FINISHED”.
12977: -                  Example
13014: -                  Output
13222: - versions in the stage will be automatically moved to the “archived” stage. Only valid
13224: + versions in the stage will be automatically moved to the “archived” stage. Only
13225: - when
13226: + valid when
13244: -                    otherwise an error will be raised.
13246: +                    otherwise an error will be
13247: + raised.
13258: -                  A single
13259: +                   A single
13266: -                     mlflow.entities.model_registry.ModelVersion
13267: +                      mlflow.entities.model_registry.ModelVersion
13274: -                  object.
13275: +                   object.
13285: -                  Example
13286: +                     Example
13515: -                  A single
13516: +                   A single
13523: -                     mlflow.entities.model_registry.ModelVersion
13524: +                      mlflow.entities.model_registry.ModelVersion
13531: -                  object.
13532: +                   object.
13542: -                  Example
13543: +                     Example
13967: -                    – The new status of the run to set, if specified.
13968: +                    – The new status of the run to set, if specified. At least one of
13970: - At least one of
13990: -                    – The new name of the run to set, if specified.
13991: +                    – The new name of the run to set, if specified. At least one of
13993: - At least one of






List of all files with diffs:
------------------------------
/Users/michael.berk/dev/mlflow/google-docs/batch_4/mlflow/docs/build/html/python_api/mlflow.html
/Users/michael.berk/dev/mlflow/google-docs/batch_4/mlflow/docs/build/html/python_api/mlflow.client.html
```

File: mlflow/tracking/_tracking_service/utils.py [hunk 4]
```diff
@@ -66,6 +64,7 @@ def set_tracking_uri(uri: Union[str, Path]) -> None:
 
         Current tracking uri: file:///tmp/my_tracking
     """
+    # pylint: enable=line-too-long
```
Comment (serena-ruan) [inline mlflow/tracking/_tracking_service/utils.py:67]: Why do we need this?

File: mlflow/tracking/fluent.py [hunk 5]
```diff
@@ -123,7 +125,7 @@ def set_experiment(
         Experiment_id: 1
         Artifact Location: file:///.../mlruns/1
         Tags: {}
-        Lifecycle_stage: active
+        Lifecycle_stage: activ
```
Comment (harupy) [inline mlflow/tracking/fluent.py:128]:
```suggestion
        Lifecycle_stage: active
```

File: mlflow/tracking/_tracking_service/utils.py (hunk 4 above)
Comment (michael-berk) [reply mlflow/tracking/_tracking_service/utils.py]: The bullets in the above docstring exceeded line length. I disabled pylint prior to docstring and reenable after. 

Wrapping the bullets solves the issue. Thanks for the spot!

**Comment by michael-berk** [comment]
@mlflow-automation autoformat

(I can't rerun failed jobs anymore)

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**